cmd/jaeger/internal/integration/e2e/grpc_test.go

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package e2e
+
+import (
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+
+	"github.com/jaegertracing/jaeger/cmd/remote-storage/app"
+	"github.com/jaegertracing/jaeger/pkg/config"
+	"github.com/jaegertracing/jaeger/pkg/healthcheck"
+	"github.com/jaegertracing/jaeger/pkg/metrics"
+	"github.com/jaegertracing/jaeger/pkg/tenancy"
+	"github.com/jaegertracing/jaeger/plugin/storage"
+	"github.com/jaegertracing/jaeger/plugin/storage/integration"
+	"github.com/jaegertracing/jaeger/ports"
+)
+
+type GRPCStorageIntegration struct {
+	E2EStorageIntegration
+
+	logger         *zap.Logger
+	server         *app.Server
+	storageFactory *storage.Factory
+}
+
+func (s *GRPCStorageIntegration) initialize() error {
+	err := s.startServer()
+	if err != nil {
+		return err
+	}
+
+	s.Refresh = func(_ *testing.T) {}
+	s.CleanUp = s.cleanUp
+	return nil
+}
+
+func (s *GRPCStorageIntegration) startServer() error {
```

Comment:
This again feels like duplication of that the existing grpc_test is already doing. From my perspective, the only differences in the e2e test case are these:
  * it needs to have jaeger-v2 process running (in-process or as binary or as container - see the other comment)
  * it needs to use RPC proxies for SpanReader/Writer instead of in-process function calls